# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientSource.cpp
# Commit: 5b8e7e4c3365
# Full Hash: 5b8e7e4c3365f09068af879fbf4d29781b2f71fb
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2020-06-04 22:00:12
# Description:
#   Bug 1643051 - Change the dignostic assertion to regular assertion for the ClientSource.cpp. r=dimi
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78264
# ==============================================================================

diff -r 95d399cee033 -r 5b8e7e4c3365 dom/clients/manager/ClientSource.cpp
--- a/dom/clients/manager/ClientSource.cpp	Thu Jun 04 11:18:07 2020 +0000
+++ b/dom/clients/manager/ClientSource.cpp	Thu Jun 04 12:56:51 2020 +0000
@@ -262,10 +262,9 @@
   // continue to inherit the SW as well.  We need to avoid triggering the
   // assertion in this corner case.
   if (mController.isSome()) {
-    MOZ_DIAGNOSTIC_ASSERT(spec.LowerCaseEqualsLiteral("about:blank") ||
-                          StringBeginsWith(spec, NS_LITERAL_CSTRING("blob:")) ||
-                          StorageAllowedForWindow(aInnerWindow) ==
-                              StorageAccess::eAllow);
+    MOZ_ASSERT(spec.LowerCaseEqualsLiteral("about:blank") ||
+               StringBeginsWith(spec, NS_LITERAL_CSTRING("blob:")) ||
+               StorageAllowedForWindow(aInnerWindow) == StorageAccess::eAllow);
   }
 
   nsPIDOMWindowOuter* outer = aInnerWindow->GetOuterWindow();
